Handling of Files with Open Trace Button

ngondalia opened the pull request:
This commit adds a handler that manages and detects resource type dialog that should be opened when opening a trace. It provides the ability to handle one or both file and folder trace types. If both is handled, a quick pick is shown to select which type of resource dialog should be opened. The functionality to control which trace type should be handle is provided through
the external API.

Signed-off-by: Neel Gondalia <ngondalia@blackberry.com>

ngondalia commented:
@bhufmann This is the follow up commit for handling of trace files with "Open Trace" button. 

In this commit, i decided to use a quick pick to select if it should be a file or folder dialog when both trace types are supported.
When only one is set, the TraceExplorerResourceTypeHandler detects which one is set and shows the appropriate dialog.

The other option is to have two separate buttons hooked to two separate commands that can be conditionally rendered (which we discussed in #216 ). 

I don't have a preference to which option, but let me know if you do and i can change whatever is needed.

Thanks! :)

bhufmann commented:
@ngondalia  I'm going to review this PR. Just a heads-up, I just merged a code clean-up patch #223 that causes merge conflicts (e.g. I renamed trace-tree.ts) with PR. You'll have to rebase and resolve the merge conflicts.

ngondalia commented:
> @ngondalia I'm going to review this PR. Just a heads-up, I just merged a code clean-up patch #223 that causes merge conflicts (e.g. I renamed trace-tree.ts) with PR. You'll have to rebase and resolve the merge conflicts.

Rebased!

bhufmann reviewed (changes requested):
It looks good to me. I ran it by some of my comments and we ok with this approach. I'd like to @hriday-panchasara to comment as well, since he was looking into that feature, too.

I have on comment though to fix.

hriday-panchasara commented on vscode-trace-extension/src/utils/trace-explorer-resource-type-handler.ts:
```diff
@@ -0,0 +1,82 @@
+        { label: 'Folder', type: 'Folder' }
+    ];
+
+    private constructor() {}
```
Getting ESLint "no-empty-function" error. You can resolve this by adding an empty comment on purpose

` private constructor() { /** Empty constructor */ }`

ngondalia replied:
Ok, fixed!

hriday-panchasara commented on vscode-trace-extension/src/trace-explorer/trace-utils.ts:
```diff
@@ -35,11 +35,11 @@ export const zoomHandler = (hasZoomedIn: boolean): void => {
 };
 
-export const openDialog = async (): Promise<vscode.Uri | undefined> => {
+export const openDialog = async (selectFiles: boolean = false): Promise<vscode.Uri | undefined> => {
```
Type boolean inferred from boolean literal. Can change to:
`selectFiles = false`

ngondalia replied:
Fixed!

hriday-panchasara reviewed (approved):
This approach works for me as well. But we should adjust the Theia implementation to use a dropdown pick or quick pick in order to have some consistency.

ngondalia commented:
> This approach works for me as well. But we should adjust the Theia implementation to use a dropdown pick or quick pick in order to have some consistency.

Yeah - I agree!

bhufmann reviewed (approved):
Looks good to me. Thanks for the contribution!

bhufmann commented:
> > This approach works for me as well. But we should adjust the Theia implementation to use a dropdown pick or quick pick in order to have some consistency.
> 
> Yeah - I agree!

@ngondalia please let me know if you going to work on this update. Theia has the `QuickPickService` that can be used for that (`@theia/core/lib/common/quick-pick-service`).

hriday-panchasara reviewed (approved):
Thanks for the contribution!
